code/game/machinery/requests_console.dm

```diff
@@ -87,38 +100,19 @@ GLOBAL_LIST_EMPTY(allConsoles)
 	. = ..()
 	name = "\improper [department] requests console"
 	GLOB.allConsoles += src
-	switch(departmentType)
-		if(1)
-			if(!("[department]" in GLOB.req_console_assistance))
-				GLOB.req_console_assistance += department
-		if(2)
-			if(!("[department]" in GLOB.req_console_supplies))
-				GLOB.req_console_supplies += department
-		if(3)
-			if(!("[department]" in GLOB.req_console_information))
-				GLOB.req_console_information += department
-		if(4)
-			if(!("[department]" in GLOB.req_console_assistance))
-				GLOB.req_console_assistance += department
-			if(!("[department]" in GLOB.req_console_supplies))
-				GLOB.req_console_supplies += department
-		if(5)
-			if(!("[department]" in GLOB.req_console_assistance))
-				GLOB.req_console_assistance += department
-			if(!("[department]" in GLOB.req_console_information))
-				GLOB.req_console_information += department
-		if(6)
-			if(!("[department]" in GLOB.req_console_supplies))
-				GLOB.req_console_supplies += department
-			if(!("[department]" in GLOB.req_console_information))
-				GLOB.req_console_information += department
-		if(7)
-			if(!("[department]" in GLOB.req_console_assistance))
-				GLOB.req_console_assistance += department
-			if(!("[department]" in GLOB.req_console_supplies))
-				GLOB.req_console_supplies += department
-			if(!("[department]" in GLOB.req_console_information))
-				GLOB.req_console_information += department
+
+	if(departmentType)
+
+		if((departmentType & REQ_DEP_TYPE_ASSISTANCE) && !("[department]" in GLOB.req_console_assistance))

```

Comment:
why are you checking "[department]" here but adding raw, is it text or object

Reply:
Silly me, I can just check raw cause it's always text